apps/worker/src/app/workflow/services/metric-queue-active.service.ts

```diff
@@ -0,0 +1,115 @@
+import { WorkerOptions } from 'bullmq';
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { QueueService } from '@novu/application-generic';
+import { checkinForCronJob } from '../../shared/utils/cronHealth';
+import * as process from 'process';
+
+const LOG_CONTEXT = 'MetricQueueService';
+const METRIC_JOB_ID = 'metric-job';
+
+@Injectable()
+export class MetricQueueActiveService extends QueueService<Record<string, never>> {
+  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
+    super('metric-active');
+
+    this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());
+
+    this.bullMqService.worker.on('completed', async (job) => {
+      checkinForCronJob(process.env.ACTIVE_CRON_ID);
+      Logger.verbose('Metric job Completed', job.id, LOG_CONTEXT);
+    });
+
+    this.bullMqService.worker.on('failed', async (job, error) => {
+      Logger.verbose('Metric job failed', LOG_CONTEXT, error);
+    });
+
+    this.addToQueueIfMetricJobExists();
+  }
+
+  private addToQueueIfMetricJobExists(): void {
+    Promise.resolve(
+      this.bullMqService.queue.getRepeatableJobs().then((job): boolean => {
+        let exists = false;
+        for (const jobElement of job) {
+          if (jobElement.id === METRIC_JOB_ID) {
+            exists = true;
+          }
+        }
+
+        return exists;
+      })
+    )
+      .then(async (exists): Promise<void> => {
+        Logger.debug('metric job exists: ' + exists, LOG_CONTEXT);
```

Comment:
maybe we should add it to the if-else statement below because there is a case where we log 'metric job exists' , and then after it log `metricJob doesn't exist, creating it`